Release 1.13 timeline and schedule

AishSundar opened the pull request:
Here's a straw-man PR for the 1.13 release schedule. In general the timeline across the calendar quarter for this release is highly crunched due to 2 Kubecons (Shangai, Nov 13-15 and Seattle, Dec 11-13), US Thanksgiving (Nov, 22-23) and the holiday season during last 2 weeks of December.  

I have highlighted some main points below in an attempt to explain my rationale around the schedule and welcome any input/feedback:

* The release cycle length is only **10 weeks** long (reasons stated above).
*  Total feature planning and development time are reduced by a week each (both are in keeping with release 1.9, which was also a shorter cycle).
    * Start of release cycle to Feature freeze is **3 weeks** vs traditional 4 weeks.
    * Time between Feature freeze and Code freeze is **3.5 weeks** vs traditional 4.5 weeks.
* Code slush and Code freeze will continue to follow a shortened timeframe, of 1 week and 2 weeks respectively, in keeping with 1.11 and 1.12 cycles. But if CI signal is not green as we get toward the intended slush and freeze we may need to elongate them by starting them earlier. This would be communicated early and broadly if it is an incoming risk to the release stability mid-cycle.
* Code thaw is **only 3 days long**! This is mainly to get the release out before Kubecon Seattle which falls in the 2nd week of December. Releasing post Kubecon (3rd week of December) will be risky due to contributors' potential unavailability and is also the least preferred option in terms of media attention and PR. Hence, aiming for a release in the 1st week of December will still give us some wiggle room to release during Kubecon in the event of possible slip.  

With all that said, **feature load and planning by SIGs will play the most crucial factor in keeping this release on track**. Delivering features on time **with quality** will help stabilize CI signal faster and release on time. During the feature definition phase we will meet with SIGs to re-emphasize release cycle length and encourage them to gauge and plan their feature aspirations accordingly. We carry on the past intent to have initial docs drafts as well as test cases arriving in conjunction with the code at or ahead of code freeze.

In the final days of the cycle in an attempt to thaw and stabilize the release sooner we may postpone more incoming cherry picks to the point release depending on risk assessments.

AishSundar commented:
/hold

AishSundar commented:
/cc @kubernetes/sig-release-members

Sig leads:
/cc Jaice Singer DuMars caleb miles

1.12 release team leads
/cc @tpepper @justaugustus @mohammedzee1000 @cjwagner @dougm @guineveresaenger @zparnold @feiskyer @nickchase @kbarnard10 

1.13 release team delta vs 1.12
/cc @jberkus @kacole2 @nikopen @tfogo @marpaia @aleksandra-malinowska

AishSundar commented:
/milestone v1.13

nickchase commented:
Maybe I'm nuts for suggesting this, but because it's such a short cycle, and because we will have ~40 new features in 1.12, maybe 1.13 should just focus on stabilizing existing features...?

AishSundar commented:
@nickchase @idvoretskyi I completely agree with you. Infact @justaugustus suggesting the same idea of a stabilization release restricting only to features falling out of 1.12. I am not sure who should to run this by for approval before we make that official. @jdumars and @calebamiles do let us know what you think.

justaugustus commented on releases/release-1.13/release-1.13.md:
```diff
@@ -0,0 +1 @@
+Please see [README.md](README.md) instead
```
We won't need this file, as no references to it have been created yet.

justaugustus commented:
Let's make sure we bring this to next SIG Release meeting, Aish.

nikhita commented on releases/release-1.13/README.md:
```diff
@@ -0,0 +1,139 @@
+* [CI Signal Report](http://bit.ly/k8s113-cisignal)
+* [Retrospective Document](http://bit.ly/k8s113-retro)
+
+**tl;dr** The 1.13 release cycle begins on **Monday, October 1st, 2018**, and ends on release day, **Thursday, December 06, 2018**.  Feature freeze is **Tuesday, October 23, 2018**.  Code slush is **Friday, November 09, 2018**. Code freeze begins **Friday, November 16, 2018** and ends **Monday, December 03, 2018**, but [may vary depending on CI Signal](#conditionally-shortened-code-freeze). Docs  ...
```
Maybe I'm getting cross-eyed between the table and the paragraph but the table shows as the release day as Jan 6, right?

nikhita replied:
Ah, found it -- the dates from code freeze upto `1.13.0-rc.1 released from branch` should be in Nov instead of Dec.

And the dates in Jan should be Dec. :)

tpepper commented on releases/release-1.13/README.md:
```diff
@@ -0,0 +1,139 @@
+| Start of Release Cycle | Lead | 1| | | | week 1 | |
+| Finalize Schedule | Lead | 5 | | | | | |
+| Begin collecting planned work from SIGs | Lead, Enhancements Lead | 8| | | | week 2 | |
+| Begin weekly release team meetings | Lead | 8 | | | | | |
```
I suggest starting your weeklies on the 1st, especially since there will be more need to coordinate tightly on features^H^H^Henhancements related discussions.

AishSundar replied:
done

tpepper commented on releases/release-1.13/README.md:
```diff
@@ -0,0 +1,139 @@
+| Clean up enhancements repo | Enhancements Lead | 16 | | | | | |
+| "Enhancements Freeze" begins (EOD PST) | Enhancements Lead | 23 | || | week 4 | |
+| 1.13.0-alpha.1 release | Branch Manager | 23 | | | | | [master-blocking], [master-upgrade] |
+| Blog post: what we're working on for 1.13 | Communications | | 2 | | | week 5 | |
```
This got dropped as of 1.11, and I mistakenly copy/pasted on to 1.12.  I defer to her, but unless @kbarnard10 has changed the plan back, I suspect this line can be dropped.

AishSundar replied:
Checked with @kbarnard10 offline and this task is good to be removed

tpepper commented on releases/release-1.13/README.md:
```diff
@@ -0,0 +1,139 @@
+| "Enhancements Freeze" begins (EOD PST) | Enhancements Lead | 23 | || | week 4 | |
+| 1.13.0-alpha.1 release | Branch Manager | 23 | | | | | [master-blocking], [master-upgrade] |
+| Blog post: what we're working on for 1.13 | Communications | | 2 | | | week 5 | |
+| All release-1.9 CI jobs/testgrid removed | Test Infra Lead | | 6 | | | week 6 | |
```
release-1.10

AishSundar replied:
I thought we kept the last 3 releases. so it will be 1.10, 11 and 1.2 for 1.13 cycle. 

In the 1.12 schedule, I see you are removing 1.8 jobs. So I am removing 1.9 now.

tpepper replied:
yes sorry off by one

tpepper commented on releases/release-1.13/README.md:
```diff
@@ -0,0 +1,139 @@
+| Blog post: what we're working on for 1.13 | Communications | | 2 | | | week 5 | |
+| All release-1.9 CI jobs/testgrid removed | Test Infra Lead | | 6 | | | week 6 | |
+| Create 'release-1.13' branch and begin daily branch | Branch Manager | | 6 | | | | |
+| v1.13.0-alpha.0 | Branch Manager | | 6 | | | | |
```
Not sure what this line is, but I think between create the branch and release the branch, this one's not needed?

AishSundar replied:
ah this should have been 1.14.0-alpha.0. Thanks for catching will fix it.

tpepper commented on releases/release-1.13/README.md:
```diff
@@ -0,0 +1,139 @@
+| Docs deadline - Open placeholder PRs | Docs Lead | | 8 | | | | |
+| All release-1.13 CI jobs/testgrid created | Test Infra Lead | | 9 | | | | |
+| Begin Code Slush | Bot, Lead | | 9 | | | week 6 | |
+| Begin MWF Burndown meetings | Lead | | 12 | | | week 7 | |
```
Could be sparsely attended?  I suggest you query ASAP and note privately the release team's travel plans for the two KubeCons (and anything else) so you know ahead of time who's going to be out of hand when.

AishSundar replied:
Thanks for the suggestion. Coming to think if it, igf I start weekly meetings from 10/1, I will move to MWF burndown a week earlier i.e., from Nov 5th.

tpepper replied:
I'd reckon you're probably ok starting in week 7, if the milestone's work list is kept shorter and less ambitious in the feature definition phase.  And might not have too many people at KubeCon China...check and see.  Might be a non-issue.

jberkus commented on releases/release-1.13/README.md:
```diff
@@ -0,0 +1,139 @@
+| 1.13.0-beta.2 released from branch | Branch Manager | | 21 | | | | [1.13-blocking], [master-blocking], [master-upgrade] |
+| Docs complete - All PRs reviewed and ready to merge | Docs Lead | | 27 | | | week 9 | |
+| 1.13.0-rc.1 released from branch | Branch Manager | | 27 | | | | [1.13-blocking], [master-blocking], [master-upgrade] |
+| End of code freeze (EOD PST): Master branch re-opens for 1.13 | Bot, Branch Manager, Lead | | 28 | | | | |
```
I'm concerned that 48 hours for code thaw-to-final-cherrypick is about 24 hours too short.  We need enough time for folks in multiple geos to review/authorize any trailing cherrypicks or fixes.

AishSundar replied:
I agree Josh that we might need a day more but I would like to aim for a shorter thaw and CP period and extend it in the worst case. With the current schedule there is almost a week of buffer(before Kubecon), thus giving us room to extend thaw by a day if need be.

AishSundar commented:
@tpepper I have addressed your comments. PTAL

All, I would like to push this draft of the schedule by Thursday 9/27, in time for 1.13 on Monday 10/1. So please provide any feedback by then. Thanks.

tpepper commented:
/lgtm

I'm curious to see if any of the broader SIG leads have any worries, but it seems like a reasonable set of compromises given everything on the calendar this last quarter of 2018!

spiffxp commented on releases/release-1.13/README.md:
```diff
@@ -0,0 +1,138 @@
+| Docs deadline - Open placeholder PRs | Docs Lead | | 8 | | | | |
+| All release-1.13 CI jobs/testgrid created | Test Infra Lead | | 9 | | | | |
+| Begin Code Slush | Bot, Lead | | 9 | | | week 6 | |
+| All Issues & PRs must have complete labels | Bug Triage | | 12 | | | week 7 | |
```
I know it's not technically release related, but I wonder if it would be worth explicitly adding some rows in here to remind people when the following are happening:
- Kubecon Shanghai (Nov 13-15)
- US Thanksgiving (Nov 22)
- Contributor summit + Kubecon Seattle (Dec 9-13)

AishSundar replied:
done

jberkus commented:
/lgtm

cjwagner commented:
bump for tide?

AishSundar commented:
@spiffxp added the Kubecons and Thanksgiving into the schedule.

@tpepper @jberkus @spiffxp need LGTM again plz

jberkus commented:
/lgtm

spiffxp commented:
/lgtm

jdumars commented:
/lgtm

tpepper commented:
/lgtm

AishSundar commented:
/hold cancel

Thanks all for the review, getting this in now
